source/blog/2015-05-05-glimmer-merging.md

```diff
@@ -0,0 +1,155 @@
+  encounter when testing Glimmer over the next few weeks will be addressed
+  before the final release.
+* The compatibility layer is quite slow in some areas, making the
+  overall Glimmer engine slower than we would like. We have plans to
+  improve overall performance through the canary and beta cycle, and
+  then of course in the 2.0 release cycle.
+* **In general, the Glimmer effort was an attempt to improve
+  re-rendering performance, especially in large lists.** It also laid
+  the groundwork for significant performance work in initial render and
+  throughout the framework, but that work is not yet done. Expect to see
+  continued performance improvements in Ember throughout the 2.x cycle
+  as a result of this change.
+
+## Performance Gains
+
+The biggest performance gains in Glimmer come from moving to a simpler
+rendering model, built on top of HTMLBars.
+
+First, this allowed us to remove **all** internal views for constructs
+like `{{foo}}`, `{{#if bar}}` and even in `{{#each posts as |post|}}`.
+This view removal has an impact on initial render, because these
+constructs are very common in real-world template.
+
+Second, as we have discussed extensively, this allows us to
+significantly improve the performance of re-rendering, which makes it
+practical to re-render lists with entirely new arrays with very good
+performance. Previously, achieving reasonable performance was very
+difficult, and even when possible, came with significant bookkeeping
+overhead.
+
+Interestingly, we have found that when testing real apps, the
+performance gains are much more widespread than we expected, in large
+part because of the simplification of the overall model.
+
+**Glimmer's performance in real applications has pleasantly surprised
+us, exceeding the improvements we've seen in benchmarks designed to
+stress-test pathological cases.**
+
+When upgrading to Glimmer, please pay special attention to the
+real-world performance of your application **in production mode** and
+after clearing any deprecations with performance warnings.
+
+## Deprecated Features
+
+Throughout the 1.x series, Ember has deprecated features we intended to
+remove in 2.0. That process has continued with Ember 1.13, which will
+contain Glimmer.
+
+However, it's worth noting that Glimmer is the first major change to
+many parts of the view since Ember was called SproutCore 2! As a result,
+perfect compatibility, especially in private features, was more
+challenging.
+
+During the process of building Glimmer, we found various semantics of
+"controller"s to be the most challenging. For the most part, this is
+because the concept of controller has a number of different meanings
+depending on context (routes, `{{render}}`, `{{#each posts
+itemController='post'}}`, `{{#each posts itemController='post' as
+|post|}}`, `{{#with someController}}`, etc.).
+
+Controllers both controller the template's "context" (called "self" in
```

Comment:
"Controllers both controller" is probably not what is meant. Maybe "Controllers both control"?